pandas/core/strings.py

```diff
@@ -2180,9 +2096,13 @@ def cat(self, others=None, sep=None, na_rep=None, join=None):
 
         # concatenate Series/Index with itself if no "others"
         if others is None:
-            result = str_cat(data, others=others, sep=sep, na_rep=na_rep)
-            return self._wrap_result(result,
-                                     use_codes=(not self._is_categorical))
+            data = data.astype(object).values
```

Comment:
why is this astype needed?

Reply:
I used it because data may be categorical, and then values is not necessarily a numpy array. Changed to `ensure_object` which you mentioned below, hope this is better.